### PR TITLE
Add machineclasses CRD

### PR DESCRIPTION
**Pull request by enxebre**
Add machineclasses CRD
`kustomize build config/default/` from https://github.com/kubernetes-sigs/cluster-api commit `0734939e05aeb64ab198e3feeee8b4e90ee5cbb2`
Related:
https://github.com/openshift/cluster-api-provider-aws/pull/116
https://github.com/openshift/installer/pull/749

### CODE REVIEW

File: install/0000_50_machine-api-operator_06_machineclass.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  creationTimestamp: null
```
Comment (frobware) [inline install/0000_50_machine-api-operator_06_machineclass.crd.yaml]: Should this be here? What's the default without specifying `null`?

File: install/0000_50_machine-api-operator_05_cluster.crd.yaml [hunk 2]
```diff
@@ -56,6 +55,12 @@ spec:
                 value:
                   type: object
                 valueFrom:
+                  properties:
```
Comment (ingvagabund) [inline install/0000_50_machine-api-operator_05_cluster.crd.yaml]: Did not know the machineclass is for cluster object as well.

**Pull request merged**